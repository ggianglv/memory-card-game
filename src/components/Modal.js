import React from "react";
import "../styles/modal.css";

const Modal = ({ onClickClose, children, title }) => {
  return (
    <div className="modal show">
      <div className="modal-content">
        <div data-testid="title" className="modal-title">{title}</div>

        <div data-testid="children" className="content-body">{children}</div>

        <div className="content-footer">
          <div onClick={onClickClose} className="close-button">
            Close
          </div>
        </div>
      </div>
    </div>
  );
};

export default Modal;
