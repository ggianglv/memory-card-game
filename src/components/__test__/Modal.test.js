import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Modal from "../Modal";
import renderer from "react-test-renderer";

test("render Modal without crash", () => {
  render(<Modal />);
});

test("render correct Modal", () => {
  const { getByTestId } = render(
    <Modal title="this is title">this is children</Modal>
  );

  expect(getByTestId("title")).toHaveTextContent("this is title");
  expect(getByTestId("children")).toHaveTextContent("this is children");
});

test("Modal to match snapshot", () => {
  const tree = renderer.create(<Modal />).toJSON();

  expect(tree).toMatchSnapshot();
});
