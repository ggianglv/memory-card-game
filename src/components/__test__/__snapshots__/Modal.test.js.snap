// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal to match snapshot 1`] = `
<div
  className="modal show"
>
  <div
    className="modal-content"
  >
    <div
      className="modal-title"
      data-testid="title"
    />
    <div
      className="content-body"
      data-testid="children"
    />
    <div
      className="content-footer"
    >
      <div
        className="close-button"
      >
        Close
      </div>
    </div>
  </div>
</div>
`;
